Sources/Authentication/AppAuthSession.swift

```diff
@@ -70,16 +72,24 @@ public final class AppAuthSession: LoginSession {
         }
     }
     
-    
+    /// Finishes the login dialog
+    ///
+    /// - Parameters:
+    ///     - callback: the URL to an authorization endpoint you started the flow with.
+    ///     - endpoint: the token endpoint to query to receive tokens back as the final action in the flow.
     @MainActor
-    public func finalise(callback url: URL) async throws -> TokenResponse {
+    public func finalise(callback url: URL, endpoint: URL) async throws -> TokenResponse {
         flow?.resumeExternalUserAgentFlow(with: url)
         
-        guard let authorizationCode else {
-            throw LoginError.inconsistentStateResponse
+        if let error {
+            throw error
```

Comment:
@Oliver-Binns worth mentioning no error was thrown if received from `present` method above, that is thrown now and tests reflect for the inconsistent state error

Reply:
What is this error? Ideally we don't want to throw any `OIDAuthorizationError` types outside of this package, we should wrap them if we can.

Reply:
@Oliver-Binns
FYI this works for bringing out the error for state mismatch but doesn't feel robust:
```swift
if let error {
     if error.localizedDescription.contains("State mismatch") {
         throw LoginError.inconsistentStateResponse
     }
}
```
🤷‍♂️

Reply:
Agree that we must wrap these errors. 

The question is how to communicate something useful to consumers. 

It seems that the AppAuth errors are hard to parse / handle and as Dylan says it feels flakey to couple to strings like "State mismatch". 

It will also be somewhat difficult to work out what other errors we can expect (although in time we'll be able to use the Auth Stub to manufacture some error scenarios and work out what AppAuth throws...)

@Oliver-Binns do you have any suggestions here?

Reply:
FYI working with Amish earlier to think about how we can wrap errors received from AppAuth produced a working implementation in the form:
```swift
switch error {
case .some(let error as NSError) where error.code == OIDErrorCodeOAuth.clientError.rawValue:
    throw LoginError.inconsistentStateResponse
case .some(let error):
    throw error
case .none:
    break
}
```
Which still leaves open the necessity for wrapping the multiple other errors AppAuth could return as a result of using the `OIDAuthState.authState` API as mentioned [in the comment left below](https://github.com/govuk-one-login/mobile-ios-authentication/pull/14#issuecomment-1822854196)

Reply:
Having a look at this, I think there's a utility provided in `App Auth` for transforming `OID Errors` into standard `NSErrors` called `OIDErrorUtilities`. Might be worth looking into using it.